File: 2364. Count Number of Bad Pairs/Solution2364.py
from typing import List


class Solution2364:
    def countBadPairs(self, nums: List[int]) -> int:
        count_map = {}
        n, res = len(nums), 0
        for i in range(n):
            dif = nums[i] - i
            res += count_map.get(dif, 0)
            count_map[dif] = count_map.get(dif, 0) + 1
        return n * (n - 1) // 2 - res
        

File: 2364. Count Number of Bad Pairs/Solution2364.cpp
#include<iostream>
#include<map>
#include<vector>

using namespace std;

class Solution2364 {
public:
    long long countBadPairs(vector<int>& nums) {
        map<int, long> count;
        long res = 0L;
        long n = (long) nums.size();
        for (int i = 0; i < n; i++) {
            int dif = nums[i] - i;
            res += count[dif]++;
            // count[dif]++;
        }
        return n*(n-1) / 2 - res;
    }
};